### PR TITLE
Fold schedules into job spec model.

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
This patch allows to specify schedules when creating job specifications.
Schedules are always embedded in job spec queries.

JIRA issues: DCOS_OSS-5970

### CODE REVIEW

File: api/src/main/resources/public/api/v1/schema/jobspec.schema.json [hunk 1]
```diff
@@ -26,6 +26,9 @@
         "type": "string"
       }
     },
+    "schedules": {
+      "type": "array"
```
Comment (jeschkies) [inline api/src/main/resources/public/api/v1/schema/jobspec.schema.json:30]: I thought we could add
```
"items": { "$ref": "schedulespec.schema.json" }
```

However, our validation does not support these kind of references.

Comment (timcharper) [reply api/src/main/resources/public/api/v1/schema/jobspec.schema.json:30]: Darn; does a newer version support it?

If not, can we at least copy-pasta?

Comment (jeschkies) [reply api/src/main/resources/public/api/v1/schema/jobspec.schema.json:30]: > Darn; does a newer version support it?

I don't quite understand the Metronome implementation. I don't think the JSON schema expects schemas to be loaded from custom paths but via URIs.

> If not, can we at least copy-pasta?
I'd say so.

File: api/src/main/scala/dcos/metronome/api/Binders.scala [hunk 2]
```diff
@@ -49,6 +49,7 @@ object Binders {
         params: Map[String, scala.collection.Seq[String]]
     ): Option[Either[String, Set[Embed]]] = {
       val embeds = params.getOrElse(key, Seq.empty).flatMap(_.split(","))
+      // TODO: marks schedules as deprecated.
```
Comment (jeschkies) [inline api/src/main/scala/dcos/metronome/api/Binders.scala:52]:
```suggestion
```

**Pull request merged**